.eslintrc.js

```diff
@@ -34,6 +34,87 @@ module.exports = {
     'ember/no-empty-attrs': 2,
     'ember/no-jquery': 2,
     'ember/no-observers': 2,
+
+    'ember/order-in-models': [
+      2,
+      {
+        order: ['attribute', 'relationship', 'single-line-function', 'multi-line-function'],
+      },
+    ],
+
+    'ember/order-in-components': [
+      2,
+      {
+        order: [
+          'service',
+          'property',
+          'empty-method',
+          'single-line-function',
+          'multi-line-function',
+          'observer',
+          'init',
+          [
+            // all the members of this array will be treated equally
+            'didReceiveAttrs',
+            'willRender',
+            'willInsertElement',
+            'didInsertElement',
+            'didRender',
+            'didUpdateAttrs',
+            'willUpdate',
+            'didUpdate',
+          ],
+          'method',
+          'actions',
+          ['willDestroyElement', 'willClearRender', 'didDestroyElement'], // all the members of this array will be treated equally
+        ],
+      },
+    ],
+
+    'ember/order-in-controllers': [
+      2,
+      {
+        order: [
+          'query-params',
+          'controller',
+          'service',
+          'inherited-property',
+          'empty-method',
+          'property',
+          'single-line-function',
+          'multi-line-function',
+          'observer',
+          'method',
+          'actions',
+        ],
+      },
+    ],
+
+    'ember/order-in-routes': [
+      2,
+      {
+        order: [
+          'service',
+          'inherited-property',
+          'property',
+          'single-line-function',
+          'multi-line-function',
+          'beforeModel',
+          'model',
+          'afterModel',
```

Comment:
is this the default order? if I recall correctly, `redirect` is fired around the same time as `afterModel`...might be nice to have them in the order they are fired...but no biggie

Reply:
- It isn't defined here by the order of execution, I took the ordering from the plugin's doc page since it looked convenient to group by functionality & then by order of execution.

- You're right about `redirect` and `afterModel` firing at the same time (https://github.com/emberjs/ember.js/blob/v3.4.1/packages/ember-routing/lib/system/route.js#L1473)